### PR TITLE
Replaces Clarivate SOAP API with REST API.

### LINKED ISSUE #1609: Switch to WoS REST API : SOAP API sunset November 2023
@lwrubel noticed this alert in HoneyBadger:  https://app.honeybadger.io/projects/50046/faults/79686845

> Savon::SOAPFault: (soap:Server) The Web of Science SOAP API is scheduled to be sunset on November 1, 2023. All users must update their code to use the Web of Science REST API. For more details and help, see https://support.clarivate.com/ScientificandAcademicResearch/s/article/Web-of-Science-SOAP-API-Sunset-Frequently-asked-questions-FAQs?language=en_US.

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made?
SOAP is so 1990s.

Fixes #1609

Requires https://github.com/sul-dlss/puppet/pull/9661/files

## How was this change tested?
Fingers crossed.

See #1618 for testing strategies

### CODE REVIEW

File: lib/web_of_science/queries.rb [hunk 1]
```diff
@@ -3,197 +3,32 @@
...
-    QUERY_LANGUAGE = 'en'
-
     attr_reader :database
 
     # @param database [String] a WOS database identifier (default 'WOK')
     def initialize(database = 'WOK')
       @database = database
     end
 
-    # @param uid [String] a WOS UID
```
Comment (justinlittman) [inline lib/web_of_science/queries.rb:19]: None of these are used.

File: lib/web_of_science/queries.rb [hunk 2]
```diff
@@ -3,197 +3,32 @@
...
       raise(ArgumentError, 'uids must be an Enumerable of WOS-UID String') if uids.blank? || !uids.is_a?(Enumerable)
 
-      message = base_uid_params.merge(uid: uids)
-      WebOfScience::Retriever.new(:retrieve_by_id, message)
-    end
-
-    # Search for MEDLINE records matching PMIDs
-    # @param pmids [Array<String>] a list of PMIDs
-    # @return [WebOfScience::Retriever]
-    def retrieve_by_pmid(pmids)
```
Comment (justinlittman) [inline lib/web_of_science/queries.rb:61]: Moved inline.

File: spec/fixtures/clarivate/links_empty_results.xml [hunk 3]
```diff
@@ -1,15 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (justinlittman) [inline spec/fixtures/clarivate/links_empty_results.xml:1]: Got rid of unused fixtures.

File: lib/agent/author_address.rb [hunk 4]
```diff
@@ -21,7 +21,7 @@ def initialize(options = {})
     end
 
     def to_xml
-      @xml ||= begin
+      @to_xml ||= begin
```
[the remark this answers is not in the document]
Comment (mjgiarlo) [reply lib/agent/author_address.rb:24]: Not sure if the name of this variable matters too much, if the interface used outside of the class is always `#to_xml`. Either's fine by me.

**Pull request merged**